test/new-e2e/pkg/e2e/suite.go

```diff
@@ -429,11 +429,25 @@ func (bs *BaseSuite[Env]) providerContext(opTimeout time.Duration) (context.Cont
 //
 // [testify Suite]: https://pkg.go.dev/github.com/stretchr/testify/suite
 func (bs *BaseSuite[Env]) SetupSuite() {
-	// Do the initial provisioning
 	// In `SetupSuite` we cannot fail as `TearDownSuite` will not be called otherwise.
 	// Meaning that stack clean up may not be called.
-	// We should `reconcileEnv` here, but currently removed for this reason.
-	// Provisioning is done in `BeforeTest` instead, the first test will support the provisioning time and failures.
+	// We do implement an explicit recover to handle this manuallay.
+	defer func() {
+		err := recover()
+		if err == nil {
+			return
+		}
+
+		bs.T().Logf("Caught panic in SetupSuite, err: %v. Will try to TearDownSuite", err)
+		bs.firstFailTest = "Initial provisioiningin SetupSuite" // This is required to handle skipDeleteOnFailure
+		bs.TearDownSuite()
+
+		panic(fmt.Errorf("Forward panic in SetupSuite after TearDownSuite, err was: %v", err))
+	}()
+
+	if err := bs.reconcileEnv(bs.originalProvisioners); err != nil {
+		panic(err)
```

Comment:
Are there other panic calls inside `reconcile..` ? If not, why panicking and recovering and then repanicking again? If yes, ok with the change

Reply:
By definition you cannot know when a `panic` will happen. Could be a null pointer or whatever. In any case we want to try to clean-up. We don't want to hide the panic from `testify.Suite`, we just want to intercept it and perform some actions, but it should still bubble up to stop the tests.

The code if `err != nil` or on a panic is the same, so I think it's clearer this way